src/exchange_adapters/bitso.ts

```diff
@@ -0,0 +1,120 @@
+import { Exchange } from '../utils'
+import { BaseExchangeAdapter, ExchangeAdapter, ExchangeDataType, Ticker, Trade } from './base'
+
+export class BitsoAdapter extends BaseExchangeAdapter implements ExchangeAdapter {
+  baseApiUrl = 'https://api.bitso.com/api/v3'
+  readonly _exchangeName = Exchange.BITSO
+  // Cloudflare Inc ECC CA-3
+  readonly _certFingerprint256 =
+    '3A:BB:E6:3D:AF:75:6C:50:16:B6:B8:5F:52:01:5F:D8:E8:AC:BE:27:7C:50:87:B1:27:A6:05:63:A8:41:ED:8A'
+
+  private static readonly tokenSymbolMap = BitsoAdapter.standardTokenSymbolMap
+
+  protected generatePairSymbol(): string {
+    return `${BitsoAdapter.tokenSymbolMap.get(
+      this.config.baseCurrency
+    )?.toLowerCase()}_${BitsoAdapter.tokenSymbolMap.get(
+      this.config.quoteCurrency
+    )?.toLowerCase()}`
+  }
+
+  async fetchTicker(): Promise<Ticker> {
+    const tickerJson = await this.fetchFromApi(
+      ExchangeDataType.TICKER,
+      `ticker?book=${this.pairSymbol}`
+    )
+    return this.parseTicker(tickerJson)
+  }
+
+  async fetchTrades(): Promise<Trade[]> {
+    const tradesJson = await this.fetchFromApi(
+      ExchangeDataType.TRADE,
+      `trades?book=${this.pairSymbol}`
+    )
+    // sort order from API is chronological
```

Comment:
Is it? Since I don't see it documented on the API docs (https://bitso.com/api_info#trades), I'd rather explicitly sort by `timestamp` after parsing.